WooCommerce/WooCommerceTests/ViewModels/Receipts/ReceiptEligibilityUseCaseTests.swift

```diff
@@ -205,92 +210,133 @@ final class ReceiptEligibilityUseCaseTests: XCTestCase {
 
         // When
         let isEligible: Bool = waitFor { promise in
-            sut.isEligibleForFailedPaymentEmailReceipts(paymentGatewayID: CardPresentPaymentsPlugin.wcPay.gatewayID, onCompletion: { result in
+            sut.isEligibleForFailedPaymentEmailReceipts(paymentGatewayID: "woocommerce-stripe", onCompletion: { result in
                 promise(result)
             })
         }
 
         // Then
+        XCTAssertFalse(isEligible)
+    }
+
+    func test_meetsOrderStatusRequirement_with_completed_status_returns_true() {
+        let stores = MockStoresManager(sessionManager: .makeForTesting())
+        let plugin = SystemPlugin.fake().copy(name: "WooCommerce", version: "9.5.0", active: true)
+
+        stores.whenReceivingAction(ofType: SystemStatusAction.self) { action in
+            switch action {
+            case let .fetchSystemPlugin(_, _, onCompletion):
+                onCompletion(plugin)
+            default:
+                XCTFail("Unexpected action")
+            }
+        }
+
+        let sut = ReceiptEligibilityUseCase(stores: stores)
+
+        let isEligible: Bool = waitFor { promise in
+            sut.meetsOrderStatusRequirement(.completed) { result in
+                promise(result)
+            }
+        }
+
         XCTAssertTrue(isEligible)
     }
 
-    func test_isEligibleForFailedPaymentEmailReceipts_when_woopayments_version_is_incorrect_then_returns_false() {
-        // Given
+    func test_meetsOrderStatusRequirement_with_processing_status_returns_true() {
```

Comment:
nit: might be worth mentioning the condition when it's true (otherwise it could sound like processing status always meets the requirement) - from my understanding, it's when the WC version is above a minimum version. Same for the refunded status test case.